From 034d37c3bef9ba99a1db7344127e814a9219047a Mon Sep 17 00:00:00 2001
From: Simon Michael <simon@joyful.com>
Date: Thu, 4 Jan 2024 16:53:24 -1000
Subject: [PATCH] fix:imp:stack: build with ghc 9.8, latest stackage nightly
 (part 2)

Fix the build with GHC < 9.8.

This reapplies commit 87600e5dd, this time with the correct description.
---
 hledger/Hledger/Cli/Script.hs | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

Index: hledger-1.32.3/Hledger/Cli/Script.hs
===================================================================
--- hledger-1.32.3.orig/Hledger/Cli/Script.hs	2024-01-28 10:05:20.000000000 +0000
+++ hledger-1.32.3/Hledger/Cli/Script.hs	2024-02-28 15:28:33.604355754 +0000
@@ -3,6 +3,7 @@ A convenient module to import in hledger
 aiming to provide the most useful imports and reduce boilerplate.
 |-}
 
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE PackageImports #-}
 
 module Hledger.Cli.Script 
@@ -15,7 +16,13 @@ import Control.Concurrent as M
 import Control.Monad as M
 import Data.Char as M
 import Data.Either as M
+
 import Data.Functor as M
+#if MIN_VERSION_base(4,19,0)
+  hiding (unzip)
+#endif
+-- import Data.Functor as M (Functor, fmap, (<$), ($>), (<$>), (<&>))  -- or without CPP ? redundant import warning
+
 import Data.List as M
 import Data.Maybe as M
 import Data.Ord as M
